nearswap/tests/fee_simulation.rs

```diff
@@ -0,0 +1,282 @@
+    );
+
+    let token1 = sample_token(&root, dai(), vec![clp_contract()]);
+    // mint for liquidity providers
+    call!(
+        root,
+        token1.mint(to_va(lp1.account_id.clone()), to_yocto("1000").into())
+    )
+    .assert_success();
+    call!(
+        root,
+        token1.mint(to_va(lp2.account_id.clone()), to_yocto("1000").into())
+    )
+    .assert_success();
+    print!("Token minted for liquidators");
+    call!(
+        root,
+        nearswap.extend_whitelisted_tokens(vec![to_va(dai())])
+    );
+
+    // Pool creation by root account
+    call!(
+        root,
+        nearswap.create_pool(to_va("dai".into())),
+        deposit = to_yocto("1")
+    )
+    .assert_success();
+
+    // Register account lp1
+    call!(
+        lp1,
+        nearswap.storage_deposit(None, Some(true)),
+        deposit = to_yocto("1")
+    )
+    .assert_success();
+    // Register account lp2
+    call!(
+        lp2,
+        nearswap.storage_deposit(None, Some(true)),
+        deposit = to_yocto("1")
+    )
+    .assert_success();
+    print!("Account Registered!");
+    // Deposit near in account deposit lp1
+    call!(
+        lp1,
+        nearswap.storage_deposit(None, None),
+        deposit = to_yocto("35")
+    )
+    .assert_success();
+    // Deposit near in account deposit lp2
+    call!(
+        lp2,
+        nearswap.storage_deposit(None, None),
+        deposit = to_yocto("35")
+    )
+    .assert_success();
+
+    // Add to accounts whitelist
+    call!(
+        lp1,
+        nearswap.add_to_account_whitelist(&vec![to_va(dai())])
+    )
+    .assert_success();
+    call!(
+        lp2,
+        nearswap.add_to_account_whitelist(&vec![to_va(dai())])
+    )
+    .assert_success();
+
+    // Depositing tokens in account deposit
+    call!(
+        lp1,
+        token1.ft_transfer_call(to_va(clp_contract()), to_yocto("100").into(), None, "".to_string()),
+        deposit = 1
+    )
+    .assert_success();
+    call!(
+        lp2,
+        token1.ft_transfer_call(to_va(clp_contract()), to_yocto("100").into(), None, "".to_string()),
+        deposit = 1
+    )
+    .assert_success();
+
+    print!("Token deposited in contract account deposit");
+
+    // Adding liquidity
+    let lp1_shares = call!(
+        lp1,
+        nearswap.add_liquidity(dai(), to_yocto_str("9"), to_yocto_str("90"), U128(0)),
+        deposit = 1
+    )
+    .unwrap_json::<U128>();
+
+    let lp2_shares = call!(
+        lp2,
+        nearswap.add_liquidity(dai(), to_yocto_str("1"), to_yocto_str("10.01"), U128(0)),
+        deposit = 1
+    )
+    .unwrap_json::<U128>();
+
+    print!("Added Liquidity!");
+    // Register alice before swapping
+    call!(
+        alice,
+        nearswap.storage_deposit(None, Some(true)),
+        deposit = to_yocto("1")
+    )
+    .assert_success();
+    // Deposit near in account deposit
+    call!(
+        alice,
+        nearswap.storage_deposit(None, None),
+        deposit = to_yocto("5")
```

Comment:
we can do a bigger deposit in the previous call. No need to call it twice.

Reply:
the previous call is registration only call where user is registering itself in account deposit, while registering the contract only keeps the required balance and send back the additional funds, that's why a second call which is a deposit call

Reply:
added a helper function for this